### DISCUSSION TITLE
Getting all the errors but allowing it to parser to continue in repeating combinators (eg `repeat` or `seperated`)

### DISCUSSION THREAD

**Discussion by JCBurnside** [winnow-rs/winnow#597] [Q&A]
I am using this for my programming language and I was hoping there is a way to somehow get both the ones that pass in a repeating parser as well as the errors for the ones that fail.  eg `
let (success,failures) = combinator::repeat(1.., combinator::alt((
    case0.map(...),
    case1.map(...),
    fail.context(...),/* want to get this for the ones that fail */
)).parse_next(input)?;
` I saw `Parser::resume_after` but that just discards the errors.  Any advice on how to approach this?

**Comment by JCBurnside** [comment]
My immediate thought after waking back up is to do something similar to when I hand wrote a parser and basically have a struct that I wrap everything in that can then be returned with the errors but feels like that almost defeats the purpose of the whole result system.

**Comment by epage** [comment]
>  I saw Parser::resume_after but that just discards the errors. Any advice on how to approach this?

The intention of the error recovery system (#96) is that parsers like `Parser::resume_after` will take the error from a `Result` and store it in the [`Recoverable`](https://docs.rs/winnow/latest/winnow/stream/struct.Recoverable.html) stream for access once all of parsing is done.

Development on that has slowed because I don't yet have a concrete test bed to try out the ideas for it and it was becoming more complex than I had hoped.  I'm slowly experimenting with a new TOML parser and doing some basic error recovery without helpers like that and hope it will help provide feedback on what can be made reusable.

An alternative design is to have Error nodes in your parse results and you have a parser that can translate an inner parser's error into an Error node.

**Reply by JCBurnside** [reply]
I see.  I'll have to test the recoverable stuff more and get back to you!

**Comment by JCBurnside** [comment]
So question on the design choice/implementation of [`RecoverableParser::recoverable_parse`](https://docs.rs/winnow/latest/winnow/trait.RecoverableParser.html#tymethod.recoverable_parse). Since it returns `(Stream, Option<O>, Vec<E>)` is the first element supposed be the stream after consuming?  From what I'm noticing on version 0.7.12 is that it returns the original stream.

**Reply by epage** [reply]
Its the original stream.  This models the internals of `Parser::parse` which captures the original stream and passes that to `ParseError`.  It didn't quite make sense for us to return `Vec<ParseError<I, E>>`, so I had the function return the components for users to create their own thing, even if it is redundant.
